### PR TITLE
Fix arm64 robustness and stop running workflows in containers

### PR DESCRIPTION
**Pull request by jmhbnz**
`runs-on` format needs to follow the rest of the template, i.e. `runs-on: "['actuated-arm64-8cpu-32gb']"`.

Once we are happy with the new on demand arm64 runners we can simplify the format for `runs-on` further to do away with needing the square brackets array format.

### CODE REVIEW

**Comment by jmhbnz** [comment]
While this PR fixes the jobs not being picked up, as shown in the robustness nightly `main-arm64` logs there is a further error we need to address: https://github.com/etcd-io/etcd/actions/runs/6638926976/job/18036276045?pr=16824

> `fuse: device not found, try 'modprobe fuse' first`

Based on my bash history on our old `arm64` runners I believe we are missing a `sed -i 's/#user_allow_other/user_allow_other/g' /etc/fuse.conf` on the host to allow docker container the jobs run in to mount the fuse device, refer: https://serverfault.com/questions/943979/using-a-fuse-mounted-directory-as-a-docker-volume 

How about we take this opportunity to just get rid of running the jobs in containers because with the new runners we no longer have isolation concerns. Let me add another commit to do that.

**Comment by jmhbnz** [comment]
Ok so even running outside a docker container on the new `actuated-arm64-8cpu-32gb` host we still encouncter `fuse: device not found, try 'modprobe fuse' first`, example: https://github.com/etcd-io/etcd/actions/runs/6639189195/job/18037031182?pr=16824

I've added the required `/etc/fuse.conf` configuration change to our workflow and that still hasn't resolved the issue.  

> FUSE requires the option allow_other as a startup argument so that other users can read and write files, besides the user owning the file system. For that, you must uncomment/add the following line on the configuration file /etc/fuse.conf
> https://github.com/dsrhaslab/lazyfs#installation 

Some docs say that a reboot or reload of fuse kernel module is needed between changing `/etc/fuse.conf`.  Will continue debugging. Might also need to verify that the fuse kernel module is actually loaded/present.

@alexellis we will likely need some help getting to the bottom of this.

**Comment by serathius** [comment]
Can we revert the change of machines for robustness and revisit it when we find a solution? Having an official support of arm64 means that we should prioritize continued testing instead of machine migration. Especially since we started seeing problem with arm since addition of memberRemove failure in robustness tests https://github.com/etcd-io/etcd/actions/runs/6585933816 https://github.com/etcd-io/etcd/actions/runs/6572603206

**Comment by alexellis** [comment]
We can add in fuse into the Kernel for Firecracker if we can track down the kernel build option and have it rolled out to the server in a short period of time.

This doesn't appear to affect the main nightly arm64 build, so I think that one can stay running on actuated while we take a look.

**Comment by alexellis** [comment]
We're publishing a new image now with FUSE built into the Kernel. How do we get this PR rebuilt/re-triggered when ready?

**Comment by alexellis** [comment]
@jmhbnz the new image is in place, can you re-trigger?

**Comment by alexellis** [comment]
Here's an SSH debug session showing fuse as being available and loaded into the Kernel:

Screenshot 2023-10-25 at 21 45 54

File: .github/workflows/e2e-arm64-template.yaml [hunk 1]
```diff
@@ -12,10 +12,6 @@ jobs:
     # this is to prevent the job to run at forked projects
     if: github.repository == 'etcd-io/etcd'
     runs-on: actuated-arm64-8cpu-32gb
-    container: golang:1.21-bookworm
```
Comment (alexellis) [inline .github/workflows/e2e-arm64-template.yaml:15]: 👍 

Hopefully this will also yield an improvement

Comment (jmhbnz) [reply .github/workflows/e2e-arm64-template.yaml:15]: Certainly do, see line 26 :)

**Comment by jmhbnz** [comment]
Tests look to be passing now, many thanks for the assistance @alexellis!  I'll get this pr tidied up and ready for merge later today 🙂

**Review by serathius — APPROVED** [review]

**Comment by serathius** [comment]
@jmhbnz Can you link to a successful run?

**Comment by serathius** [comment]
nvm, found it https://github.com/etcd-io/etcd/actions/runs/6644556539

**Comment by ahrtr** [comment]
[tests-arm64-template.yaml#L20](https://github.com/etcd-io/etcd/blob/4d77fd1efa2d8158210c76dc6578cdd8c4d13119/.github/workflows/tests-arm64-template.yaml#L20) is still using container.

**Comment by jmhbnz** [comment]
> [tests-arm64-template.yaml#L20](https://github.com/etcd-io/etcd/blob/4d77fd1efa2d8158210c76dc6578cdd8c4d13119/.github/workflows/tests-arm64-template.yaml#L20) is still using container.

Fixed, thank you 🙏🏻

File: .github/workflows/robustness-template.yaml [hunk 2]
```diff
@@ -50,7 +41,8 @@ jobs:
           go clean -testcache
 
           # Build LazyFS
-          apt update && apt-get --yes install cmake libfuse3-dev libfuse3-3 fuse3
+          sudo apt update && sudo apt-get --yes install cmake libfuse3-dev libfuse3-3 fuse3
+          sudo sed -i 's/#user_allow_other/user_allow_other/g' /etc/fuse.conf
```
[the remark this answers is not in the document]
Comment (jmhbnz) [reply .github/workflows/robustness-template.yaml:45]: I did it manually on the two static `arm64` machines as a one-off historically.  

As we are now moving to ephemeral machines I believe we need to make it part of the workflow as it's a requirement stipulated by lazyfs in their install docs.

[the remark this answers is not in the document]
Comment (ahrtr) [reply .github/workflows/robustness-template.yaml:45]: Got it, thx for the clarification.

**Review by ahrtr — APPROVED** [review]
LGTM

thx @jmhbnz

**Pull request merged**